runtime/runtime/src/lib.rs

```diff
@@ -1450,6 +1450,9 @@ impl Runtime {
         // TODO(#8859): Introduce a dedicated `compute_limit` for the chunk.
         // For now compute limit always matches the gas limit.
         let compute_limit = apply_state.gas_limit.unwrap_or(Gas::max_value());
+        let check_state_witness_size_limit =
```

Comment:
nit: I would use single `Option` param here:
`let state_witness_size_limit = if checked_feature!("stable", StateWitnessSizeLimit, protocol_version) Some(apply_state.config.fees.state_witness_size_soft_limit) else None`

then you check the limit with [is_some_and](https://doc.rust-lang.org/std/option/enum.Option.html#method.is_some_and)